nifi-nar-bundles/nifi-graph-bundle/nifi-other-graph-services/pom.xml

```diff
@@ -90,6 +90,39 @@
             <artifactId>gremlin-driver</artifactId>
             <version>${gremlin.version}</version>
         </dependency>
+
+        <dependency>
+            <groupId>org.apache.groovy</groupId>
+            <artifactId>groovy</artifactId>
+            <version>${nifi.groovy.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.groovy</groupId>
+            <artifactId>groovy-jsr223</artifactId>
```

Comment:
@mattyb149 I have tried removing it, but it fails with a NPE, because it uses `javax.script.ScriptEngineManager.getEngineByName("groovy")` to compile the groovy script. See:

https://github.com/apache/nifi/pull/7677/files#diff-bade439cf2e90169d5771ee9bf9a3df690c07e80384309141e66c3aad42d422aR253

And: 

https://github.com/apache/nifi/pull/7677/files#diff-bade439cf2e90169d5771ee9bf9a3df690c07e80384309141e66c3aad42d422aR458

Do you have a recommended way to do this?

Reply:
@mattyb149  -- was able to refactor the code to remove this! Sorry if you got a notification of a previous message, some googling around got me around the issue.